src/sio.ts

```diff
@@ -117,6 +146,56 @@ export class RPSIO {
       case GPIO_HI_OE_XOR:
         this.qspiGpioOutputEnable ^= value & GPIO_MASK;
         break;
+      case DIV_UDIVIDEND:
+        this.divDividend = value >>> 0;
+        this.divCSR = 0b10;
+        this.divQuotient = Math.trunc(this.divDividend / this.divDivisor) >>> 0;
+        this.divRemainder = Math.trunc(this.divDividend % this.divDivisor) >>> 0;
+        this.divCSR = 0b11;
+        this.rp2040.cycles += 8;
+        break;
+      case DIV_SDIVIDEND:
+        this.divDividend = value | 0;
+        this.divCSR = 0b10;
+        this.divQuotient = Math.trunc(this.divDividend / this.divDivisor) | 0;
+        this.divRemainder = Math.trunc(this.divDividend % this.divDivisor) | 0;
+        this.divCSR = 0b11;
+        this.rp2040.cycles += 8;
+        break;
+      case DIV_UDIVISOR:
+        this.divDivisor = value >>> 0;
+        this.divCSR = 0b10;
+        if (this.divDivisor == 0) {
+          this.divQuotient = 0xffffffff;
+          this.divRemainder = this.divDividend;
+        } else {
+          this.divQuotient = Math.trunc(this.divDividend / this.divDivisor) >>> 0;
+          this.divRemainder = Math.trunc(this.divDividend % this.divDivisor) >>> 0;
+        }
+        this.divCSR = 0b11;
+        this.rp2040.cycles += 8;
+        break;
+      case DIV_SDIVISOR:
+        this.divCSR = 0b10;
+        this.divDivisor = value | 0;
+        if (this.divDivisor == 0) {
```

Comment:
I suggest that we move this repetitive code (in DIV_UDIVIDEND, DIV_SDIVIDEND, DIV_UDIVISOR, and DIV_SDIVISOR) into a function that will do the actual division, e.g. `this.updateDivider()`.

As far as I can tell, you don't need to worry about the sign of the result, because when it's read back into a register it'll always be internally converted to unsigned anyway.